rest/import_test.go

```diff
@@ -698,6 +698,236 @@ func TestMigrateWithExternalRevisions(t *testing.T) {
+
+	// Wait for doc to appear on changes feed
+	// Wait until the change appears on the changes feed to ensure that it's been imported by this point
+	now := time.Now()
+	changes, err := rt.WaitForChanges(1, "/db/_changes", "", true)
+	assertNoError(t, err, "Error waiting for changes")
+	changeEntry := changes.Results[0]
+	assert.Equals(t, changeEntry.ID, key)
+	log.Printf("Saw doc on changes feed after %v", time.Since(now))
+
+	// Double-check to make sure that it's been imported by checking the Sync Metadata in the xattr
+	assertXattrSyncMetaRevGeneration(t, bucket, key, 1)
+
+	// Now get the doc expiry and validate that it has been migrated into the doc metadata
+	gocbBucket := bucket.(*base.CouchbaseBucketGoCB)
+	expiry, err := gocbBucket.GetExpiry(key)
+	assert.True(t, expiry > 0)
+	assertNoError(t, err, "Error calling GetExpiry()")
+	log.Printf("expiry: %v", expiry)
+	assert.True(t, expiry == uint32(testExpiry.Unix()))
+
+}
+
+
+// Write a doc via SDK with an expiry value.  Verify that expiry is preserved when doc is imported via on-demand
+// import (GET or WRITE)
+func TestXattrOnDemandImportPreservesExpiry(t *testing.T) {
+
+	SkipImportTestsIfNotEnabled(t)
+
+	var rt RestTester
+
+	mobileBody := make(map[string]interface{})
+	mobileBody["type"] = "mobile"
+	mobileBody["channels"] = "ABC"
+
+	triggerOnDemandViaGet := func(key string) {
+		rt.SendAdminRequest("GET", fmt.Sprintf("/db/%s", key), "")
+	}
+	triggerOnDemandViaWrite := func(key string) {
+		mobileBody["foo"] = "bar"
+		mobileBodyMarshalled, err := json.Marshal(mobileBody)
+		assertNoError(t, err, "Error marshalling body")
+		rt.SendAdminRequest("PUT", fmt.Sprintf("/db/%s", key), string(mobileBodyMarshalled))
+	}
+
+	testCases := []func(string){
+		triggerOnDemandViaGet,
+		triggerOnDemandViaWrite,
+	}
+
+	for i, triggerOnDemand := range testCases {
+		t.Run(fmt.Sprintf("Subtest"), func(t *testing.T) {
+
+			rt = RestTester{
+				SyncFn:         `function(doc, oldDoc) { channel(doc.channels) }`,
+				DatabaseConfig: &DbConfig{},
+			}
+			defer rt.Close()
+			bucket := rt.Bucket()
+
+			rt.SendAdminRequest("PUT", "/_logging", `{"Import+":true, "CRUD+":true}`)
+
+			key := fmt.Sprintf("TestXattrOnDemandImportPreservesExpiry-%d", i)
+
+			// Write directly to bucket with an expiry
+			expiryUnixEpoch := time.Now().Add(time.Second * 30).Unix()
+			_, err := bucket.Add(key, uint32(expiryUnixEpoch), mobileBody)
+			assertNoError(t, err, "Error writing SDK doc")
+
+			// Verify the expiry before the on-demand import is triggered
+			gocbBucket := bucket.(*base.CouchbaseBucketGoCB)
+			expiry, err := gocbBucket.GetExpiry(key)
+			assertNoError(t, err, "Error calling GetExpiry()")
+			assert.True(t, expiry == uint32(expiryUnixEpoch))
+
+			triggerOnDemand(key)
+
+			// Wait until the change appears on the changes feed to ensure that it's been imported by this point.
+			// This is probably unnecessary in the case of on-demand imports, but it doesn't hurt to leave it in as a double check.
```

Comment:
Shouldn't be needed for on-demand, but also won't hurt.  Maybe just update the above comment to say why we're leaving it here.